### PR TITLE
[Messenger] Fix missing auto_setup for RedisTransport

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | 4.3
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

Should be my last PR for messenger 4.3's Redis transport :)
Not having it makes it inconsistent with other transports and is especially annoying in tests.

### CODE REVIEW

File: src/Symfony/Component/Messenger/Transport/RedisExt/Connection.php [hunk 1]
```diff
@@ -141,6 +159,10 @@ public function reject(string $id): void
 
     public function add(string $body, array $headers): void
     {
+        if ($this->autoSetup) {
```
Comment (alexander-schranz) [inline src/Symfony/Component/Messenger/Transport/RedisExt/Connection.php]: I think we should in some case avoid to call setup on every add or get, maybe set `$this->autoSetup` to false in setup() function?

File: src/Symfony/Component/Messenger/Transport/RedisExt/Connection.php [hunk 2]
```diff
@@ -27,6 +27,13 @@
  */
 class Connection
 {
+    private const DEFAULT_OPTIONS = [
+        'stream' => 'messages',
+        'group' => 'symfony',
+        'consumer' => 'consumer',
+        'auto_setup' => true,
+    ];
+
```
Comment (alexander-schranz) [inline src/Symfony/Component/Messenger/Transport/RedisExt/Connection.php]: missing declarion of `private $autoSetup;`

**Review by alexander-schranz — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Messenger/Transport/RedisExt/Connection.php (hunk 1 above)
Comment (chalasr) [reply src/Symfony/Component/Messenger/Transport/RedisExt/Connection.php]: 👍 done

File: src/Symfony/Component/Messenger/Transport/RedisExt/Connection.php (hunk 2 above)
Comment (chalasr) [reply src/Symfony/Component/Messenger/Transport/RedisExt/Connection.php]: fixed

**Review by alexander-schranz — APPROVED** [review]

**Review by maxhelias — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @chalasr.

**Pull request merged**